### PR TITLE
Fix Labels, Layout, and Responsive Styles for Entity List Filter Bar and Legend

### PR DESCRIPTION
**Pull request by tn3rb**
This PR does the following:

- adds labels underneath main filter bar buttons and updates tooltips

- extracts labels for filter buttons into a new shared component

- fixes some other inconsistencies with filter bar components

- adjusts layout for legend items especially for mobile

### CODE REVIEW

File: assets/src/application/ui/display/EntityListLegend/index.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const EntityListLegend: React.FC<Props> = ({ legendConfig }) => {
 
 	const iconsSource = icons.map(({ icon, description }) => {
 		return {
-			className: 'ee-legend-description',
+			className: 'ee-entity-list-legend-item',
```
Comment (tn3rb) [inline assets/src/application/ui/display/EntityListLegend/index.tsx]: this is the class name for the entire list item not just description

File: assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import Collapsible from './Collapsible';
 import { EntityListFilterBarProps } from './types';
 import { EntityListLegend } from '@application/ui/display';
 import GridViewFilterButton from './buttons/GridView';
-import LegendButton from './buttons/LegendButton';
+import ToggleLegendButton from './buttons/ToggleLegend';
```
Comment (tn3rb) [inline assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx]: just for consistency's sake

File: assets/src/application/ui/layout/entityList/filterBar/buttons/ToggleFilters.tsx [hunk 3]
```diff
@@ -2,20 +2,18 @@ import React from 'react';
 import classNames from 'classnames';
 import { __ } from '@wordpress/i18n';
 
+import FilterButtonWrap from './FilterButtonWrap';
 import { ToggleFiltersButtonProps } from '../types';
 import { EspressoButton, EspressoButtonType, Icon } from '@appInputs/EspressoButton';
 
-const ToggleFiltersButton: React.FC<ToggleFiltersButtonProps> = ({ listId, showFilters, toggleFilters }) =>
+const ToggleFilters: React.FC<ToggleFiltersButtonProps> = ({ listId, showFilters, toggleFilters }) =>
```
Comment (tn3rb) [inline assets/src/application/ui/layout/entityList/filterBar/buttons/ToggleFilters.tsx]: to match existing file name (for consistency's sake)

File: assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx (hunk 2 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/layout/entityList/filterBar/EntityListFilterBar.tsx]: Shouldn't it be `LegendToggleButton`? 🤔

**Pull request merged**